src/components/TodoComponents/TodoList.js

```diff
@@ -40,45 +47,84 @@ class TodoList extends React.Component {
     });
   };
 
-  // removeCompleted = event => {
-  //   this.setState({
-
-  //   })
-  // }
-
   updateList = event => {
     event.preventDefault();
-    console.log('page')
+
     let newItem = {
       task: this.state.task,
-      id: this.state.id,
-      completed: this.state.completed
+      id: Date.now(),
+      completed: false
     };
 
+    let exchange = this.state.taskList.slice();
+    
+    exchange.push(newItem);
+
+    stored.setItem('exchange', JSON.stringify(exchange))
+
     this.setState({
-      taskList: [...this.state.taskList, newItem]
+      taskList: JSON.parse(stored.getItem('exchange')),
+      // taskList: exchange,
+      task:''
     });
   };
 
+  toggleCompleted = event => {
+    let exchange = this.state.taskList.map(task => {
+      if (task.id === event) {
+        task.completed = !task.completed;
+        return task
+      }
+      else {
+        return task
+      };
+    })
+    this.setState({
+      taskList: exchange     
+   })   
+  }
+
+  removeComplete = event => {
+    event.preventDefault();
+
+    let exchange = this.state.taskList.filter(task => 
+      task.completed === false      
+    )
+
+    stored.setItem('exchange', JSON.stringify(exchange))
+
+    this.setState({
+      taskList: JSON.parse(stored.getItem('exchange')),
```

Comment:
this all looks awesome, great local storage code, but just remember what will's project did, it might be better to put this store in local storage in a  separate function and run it every time your page re renders.